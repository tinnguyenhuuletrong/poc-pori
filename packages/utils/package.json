{
  "name": "@pori-and-friends/utils",
  "version": "0.0.1",
  "type": "commonjs"
}
