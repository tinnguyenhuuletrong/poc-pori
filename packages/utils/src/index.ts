export * from './lib/axiosHelper';
export * from './lib/typeConvertHelper';
export * from './lib/cryptoHelper';
export * from './lib/deferred';

export const waitForMs = (ms: number) => new Promise((r) => setTimeout(r, ms));
