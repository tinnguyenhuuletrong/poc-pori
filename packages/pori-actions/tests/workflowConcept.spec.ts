import { waitForMs } from '@pori-and-friends/utils';
import {
  createWorkflow,
  FlowExec,
  WorkflowState,
} from '../src/lib/workflow/workflowV1';

describe('workflow_concept', () => {
  const exec: FlowExec = async (state: WorkflowState) => {
    const it = 3;
    let sum = 0;
    for (let i = 0; i < it; i++) {
      state.updateState(() => {
        state.currentStep = i.toString();
        sum += i;
        state.data['sum'] = sum;
      });

      await state.promiseWithAbort(waitForMs(10));
    }
  };

  test('workflow_can_finish', async () => {
    const workflow = createWorkflow(exec);

    workflow.onChange = () => {
      console.log(workflow.currentStep);
    };
    await workflow.start();
  });

  test('workflow_can_abort', async () => {
    const workflow = createWorkflow(exec);

    workflow.onChange = () => {
      if (workflow.currentStep === '1') return workflow.abort();
      console.log(workflow.currentStep);
    };
    const p = workflow.start();
    await expect(p).rejects.toThrowErrorMatchingSnapshot();
  });
});
