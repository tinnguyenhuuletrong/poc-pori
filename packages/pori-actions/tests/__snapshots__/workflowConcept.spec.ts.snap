// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workflow_concept workflow_can_abort 1`] = `"aborted"`;
